pkg/lib/cockpit-components-table.jsx

```diff
@@ -78,41 +78,32 @@ export const ListingTable = ({
     sortMethod,
     ...extraProps
 }) => {
-    let rows = tableRows;
-
+    let rows = [...tableRows];
     const [expanded, setExpanded] = useState({});
     const [newItems, setNewItems] = useState([]);
     const [currentRowsKeys, setCurrentRowsKeys] = useState([]);
     const [activeSortIndex, setActiveSortIndex] = useState(sortBy ? sortBy.index : 0);
     const [activeSortDirection, setActiveSortDirection] = useState(sortBy ? sortBy.direction : SortByDirection.asc);
+    const rowKeys = rows.map(row => row.props ? row.props.key : undefined)
+            .filter(key => key !== undefined);
+    const rowKeysStr = JSON.stringify(rowKeys);
+    const currentRowsKeysStr = JSON.stringify(currentRowsKeys);
 
     useEffect(() => {
-        const getRowKeys = rows => {
-            const keys = [];
-
-            rows.forEach(row => {
-                if (row.props && row.props.key !== undefined)
-                    keys.push(row.props.key);
-            });
-
-            return keys;
-        };
-
-        const current_keys = getRowKeys(rows);
-        if (JSON.stringify(current_keys) === JSON.stringify(currentRowsKeys))
-            return;
-
         // Don't highlight all when the list gets loaded
-        if (currentRowsKeys.length !== 0) {
-            const new_keys = current_keys.filter(key => currentRowsKeys.indexOf(key) === -1);
+        const _currentRowsKeys = JSON.parse(currentRowsKeysStr);
+        const _rowKeys = JSON.parse(rowKeysStr);
```

Comment:
I don't get this. Few lines above, we stingify the object, and then here we turn the JSON string back into the same object? Cannot we use the same rowKeys and currentRowsKeys objects?

Reply:
that's the point of this PR, if we use the objects, react get's confused and endlessly re-renders causing warnings in the console. 
Check console in any page that contains a table and you will see tons of:
 
Warning: Maximum update depth exceeded. This can happen when a component calls setState inside useEffect, but useEffect either doesn't have a dependency array, or one of the dependencies changes on every render.%s

when adding or removing elements from an array.

Reply:
I get the stringification part for using as row key, but why the `.parse()` call? I. e. why do you need to create `_rowKeys` and `_currentRowKeys` instead of just using the original values?

Reply:
> that's the point of this PR, if we use the objects, react get's confused and endlessly re-renders causing warnings in the console.

Isn't that fixed in `setCurrentRowsKeys` where you pass strings instead of the whole objects? Passing objects there causes the endless re-renders no?
Therefore I understand how does using `rowKeysStr` and `currentRowsKeysStr` change rerender. 

But how does using `_rowKeys` and `_currentRowKeys` change anything instead of just using `rowKeys` and `currentRowKeys`? Because here you use `_currentRowsKeys` and `_rowKeys` only as part of some if condition few lines below, and do not pass them to a state, so nothing changed the way rerender works, no?

Reply:
@martinpitt @skobyda if I would use the rowKeys and currentRowKeys in the useEffect body I would need to add these to in the dependency array of this hook, that cancelling the whole point of this PR, as these variable are complicated objects, and would make the code inside the useEffect hook run without changes happening.

Reply:
Sorry, I mis-read the patch. The JSON stringification happens in render, outside of the effect. And the parsing happens inside. So this makes sense.